### ISSUE TITLE
Out of Memory for batchsize=2 on A10 card

### ISSUE THREAD

**Issue by tengshaofeng** [deepglint/unicom#120]
Hi, thanks for your great job. When I use your code to finetune inshop based on "ViT-L/14@336px"， but after 8 steps， it comes to OutOfMemory. And I have only one card called A10 24G. 

#  set 
args.batch_size = 2  # 16/8，因为从8卡改为单卡
args.dataset = 'inshop'# "miaojie"
args.debug = 0
args.epochs = 128
args.lr = 1e-05
args.lr_pfc_weight = 10.0
args.input_size = 336
args.gradient_acc = 1
args.model_name = "ViT-L/14@336px"
args.margin_loss_m1 = 1.0
args.margin_loss_m2 = 0.25
args.margin_loss_m3 = 0.0
args.margin_loss_s = 32.0
args.margin_loss_filter = 0.0
args.num_workers = 4
args.num_feat = 512
args.optimizer = "adamw"
args.output_dim = 768
args.output = "/tmp/tmp_for_training"
args.resume = "NULL"
args.sample_rate = 1.0
args.seed = 1024
args.transform = "timm"
args.weight_decay = 0

**Comment by tengshaofeng** [comment]
@anxiangsir how about modifying from "with torch.cuda.amp.autocast(False)" to "with torch.cuda.amp.autocast()".
Will it be harmful to training result?

**Comment by tengshaofeng** [comment]
@anxiangsir self.weight = torch.nn.Parameter(torch.normal(0, 0.01, (self.num_local, embedding_size))) It consumes too much GPU memory，especially when nb_class=self.num_local=880000, I can not run it even set batchsize=2 based on 224
